practice/prototype_mutations_history_gc.py

```diff
@@ -0,0 +1,537 @@
+        
+        This results in self.nodes['generation'].min() 
+        equal to -0.0 (e.g, 0.0 with the negative bit set),
+        but msprime does not seem to mind.
+        """
+        self.nodes['generation'] -= self.nodes['generation'].max()
+        self.nodes['generation'] *= -1.0
+
+    def reset_data(self):
+        """
+        Call this after garbage collection.
+        Any necessary post-processing is done here.
+        """
+        self.nodes = np.empty([0], dtype=node_dt)
+        self.edges = np.empty([0], dtype=edge_dt)
+        self.mutations = np.empty([0], dtype=mutation_dt)
+
+    def post_gc_cleanup(self, gc_rv):
+        """
+        Pass in the return value from ARGsimplifier.__call__.
+
+        We clean up internal data if needed.
+        """
+        if gc_rv[0] is True:
+            self.reset_data()
+
+
+class ARGsimplifier(object):
+    """
+    Mimicking the API if our Python
+    class to collect simulated
+    results and process them via msprime
+    """
+    __nodes = None
+    __edges = None
+    __mutations = None
+    __sites = None
+    __gc_interval = None
+    __last_gc_time = None
+
+    def __init__(self, gc_interval=None):
+        self.__nodes = msprime.NodeTable()
+        self.__edges = msprime.EdgeTable()
+        self.__mutations = msprime.MutationTable()
+        self.__sites = msprime.SiteTable()
+        self.gc_interval = gc_interval
+        self.last_gc_time = 0
+
+    def simplify(self, generation, tracker):
+        """
+        Details of taking new data, appending, and
+        simplifying.
+
+        :return: length of simplifed node table, which is next_id to use
+        """
+        # Update time in current nodes.
+        # Is this most effficient method?
+        dt = generation - self.last_gc_time
+        self.nodes.set_columns(flags=self.nodes.flags,
+                               population=self.nodes.population,
+                               time=self.nodes.time + dt)
+
+        # Create "flags" for new nodes.
+        # This is much faster than making a list
+        flags = np.empty([len(tracker.nodes)], dtype=np.uint32)
+        flags.fill(1)
+
+        # Convert time from forwards to backwards
+        tracker.convert_time()
+
+        # Update internal *Tables
+        self.nodes.append_columns(flags=flags,
+                                  population=tracker.nodes['population'],
+                                  time=tracker.nodes['generation'])
+        self.edges.append_columns(left=tracker.edges['left'],
+                                  right=tracker.edges['right'],
+                                  parent=tracker.edges['parent'],
+                                  child=tracker.edges['child'])
+        self.sites.append_columns(position=tracker.mutations['position'],
+                   ancestral_state=np.zeros(len(tracker.mutations['position']),np.int8),
+                   ancestral_state_length=np.ones(len(tracker.mutations['position']),np.uint32))
```

Comment:
The `foo_length` is removed in the current msprime API.  The current master would have this line read as:

`ancestral_state_offset=np.arange(len(tracker.mutations['position'])+1,np.uint32)`

Similar for all occurrences of `derived_state_length`.